olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp

```diff
@@ -190,9 +231,66 @@ TEST_F(CatalogRepositoryTest, GetLatestVersionCacheOnlyRequestWithMinVersion) {
   EXPECT_EQ(response.GetResult().GetVersion(), kStartVersion);
 }
 
-TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyNotFound) {
-  olp::client::CancellationContext context;
+TEST_F(CatalogRepositoryTest,
+       AsyncGetLatestVersionCacheOnlyRequestWithMinVersion) {
+  auto request = read::CatalogVersionRequest();
+  request.WithFetchOption(olp::dataservice::read::CacheOnly)
+      .WithStartVersion(kStartVersion);
+
+  EXPECT_CALL(*cache_, Get(_, _))
+      .Times(1)
+      .WillOnce(testing::Return(boost::any{}));
+
+  EXPECT_CALL(*cache_, Put(_, _, _, _)).Times(1);
 
+  ON_CALL(*network_, Send(_, _, _, _, _))
+      .WillByDefault([](olp::http::NetworkRequest, olp::http::Network::Payload,
+                        olp::http::Network::Callback,
+                        olp::http::Network::HeaderCallback,
+                        olp::http::Network::DataCallback) {
+        ADD_FAILURE() << "Should not be called with CacheOnly";
+        return olp::http::SendOutcome(olp::http::ErrorCode::UNKNOWN_ERROR);
+      });
+
+  ApiLookupClient lookup_client(kHrn, settings_);
+  repository::CatalogRepository repository(kHrn, settings_, lookup_client);
+
+  olp::client::ApiResponse<read::CatalogVersionResult, olp::client::ApiError>
+      result;
+  auto response = repository.GetLatestVersion(
```

Comment:
```suggestion
  const auto response = repository.GetLatestVersion(
```

By the way, you don't use this return value. And it has the same name as an lambda argument on the line 263